### ISSUE TITLE
Problems with non indexed metadata fields - Dspace 7.6.1

### ISSUE THREAD

**Issue by aroldorique** [DSpace/DSpace#9419]
Hello everybody.

My searches in Dspace 7.6 are searching in metadata that should not be indexed. For example: the dc.description.provenance metadata. Some searches I do are returning items because the search parameter is within the item's dc.description.provenance metadata.

I am following the documentation described in this link: https://wiki.lyrasis.org/display/DSDOC7x/Discovery#Discovery-Nonindexedmetadatafields and I have included the dc.description.provenance metadata in the list of those that should not be indexed.

Below I am placing my org.dspace.discovery.configuration.DiscoveryConfigurationService bean from discovery.xml. Something's missing? Am I doing something wrong? How do I stop the dc.description.provenance metadata from being indexed?

    <bean id="org.dspace.discovery.configuration.DiscoveryConfigurationService" class="org.dspace.discovery.configuration.DiscoveryConfigurationService">
        <property name="map">
            <map>
                
                
                <entry key="default" value-ref="defaultConfiguration" />
                
                
                <entry key="default-relationships" value-ref="defaultRelationshipsConfiguration" />

                
                <entry key="site" value-ref="homepageConfiguration" />
                
                
                
                
                <entry key="workspace" value-ref="workspaceConfiguration" />
                <entry key="supervisedWorkspace" value-ref="supervisedWorkspaceConfiguration" />
                
                <entry key="workflow" value-ref="workflowConfiguration" />
                
                <entry key="workflowAdmin" value-ref="workflowAdminConfiguration" />
                
                <entry key="supervision" value-ref="supervisionConfiguration" />

                <entry key="undiscoverable" value-ref="unDiscoverableItems" />
                <entry key="administrativeView" value-ref="administrativeView" />

                <entry key="publication" value-ref="publication"/>
                
                
                <entry key="publication-relationships" value-ref="publicationRelationships"/>

                <entry key="person" value-ref="person"/>
                
                
                <entry key="person-relationships" value-ref="personRelationships"/>

                <entry key="orgunit" value-ref="orgUnit"/>
                
                
                <entry key="orgunit-relationships" value-ref="orgUnitRelationships"/>

                <entry key="journalissue" value-ref="journalIssue"/>
                
                
                <entry key="journalissue-relationships" value-ref="journalIssueRelationships"/>

                <entry key="journalvolume" value-ref="journalVolume"/>
                
                
                <entry key="journalvolume-relationships" value-ref="journalVolumeRelationships"/>

                <entry key="journal" value-ref="journal"/>
                
                
                <entry key="journal-relationships" value-ref="journalRelationships"/>

                <entry key="project" value-ref="project"/>
                
                
                <entry key="project-relationships" value-ref="projectRelationships"/>

                
                <entry key="personOrOrgunit" value-ref="personOrOrgunit"/>
                
                <entry key="openAIREFundingAgency" value-ref="openAIREFundingAgency"/>
                <entry key="eperson_claims" value-ref="eperson_claims"/>
            </map>
        </property>
        <property name="toIgnoreMetadataFields">
            <map>
                <entry>
                    <key><util:constant static-field="org.dspace.core.Constants.COMMUNITY"/></key>
                    <list>
                        
                        
                        
                        
                        
                        
                        
                        <value>dc.rights</value>
                        
                        
                    </list>
                </entry>
                <entry>
                    <key><util:constant static-field="org.dspace.core.Constants.COLLECTION"/></key>
                    <list>
                        
                        
                        
                        
                        
                        
                        
                        <value>dc.rights</value>
                        
                        
                    </list>
                </entry>
                <entry>
                    <key><util:constant static-field="org.dspace.core.Constants.ITEM"/></key>
                    <list>
                        <value>dc.description.provenance</value>
                    </list>
                </entry>
            </map>
        </property>
    </bean>

Aroldo

**Comment by alanorth** [comment]
@aroldorique sounds like a misconfiguration. Have you diffed your `discovery.xml` with the stock one to check if anything is out of place? I don't see any results for a wildcard query on the provenance field on demo.dspace.org or my own repository:

https://demo.dspace.org/search?query=dc.description.provenance%3A*

**Comment by aroldorique** [comment]
Alan, thanks for the response.

In our repository, if I search for the string "aline correia guimaraes",
some items are returned where this string is only found in the
dc.description.provenance metadata, as in the case below.

https://biblioteca.mpf.mp.br/repositorio/search?query=%22aline%20correia%20guimaraes%22&spc.page=3

[image: image.png]

Notice that one of the items returned (red arrow) is this:
http://hdl.handle.net/11549/246867

In the metadata list for this item, the string "aline correia guimaraes" is
found only in the dc.description.provenance metadata.

[image: image.png]

But, if I do this search: dc.description.provenance: "aline correia
guimaraes", no results are returned. What could be happening?
[image: image.png]

Aroldo


















Em seg., 18 de mar. de 2024 às 11:48, Alan Orth ***@***.***>
escreveu:

**Comment by aroldorique** [comment]
I am sending the three images that were not attached correctly in my previous post.

![bug1](https://github.com/DSpace/DSpace/assets/31450466/897c1bc3-a5f7-4d21-b6d3-b0b97e6d1b02)
![bug2](https://github.com/DSpace/DSpace/assets/31450466/28350a5f-9fea-4305-b66b-9497ecdad967)
![bug3](https://github.com/DSpace/DSpace/assets/31450466/30fa5853-1390-46c6-84ad-8a269f16fa9f)

**Comment by alanorth** [comment]
@aroldorique yes I see what you mean now. I don't know what is going on. I wonder if you can re-produce it on demo.dspace.org by adding some secret or unique text like "aline correia guimaraes" to the provenance field of some item.

**Comment by aroldorique** [comment]
Hello Alan.

I think I found the problem. Doing the Solr query for that item I mentioned
previously, I see that some indexed fields have the string "Aline Correia
Guimarães" (figure below). This is probably why the item is returned in the
query. I think these fields should not be indexed.

Solr Query:
https://svlp-dspace02.pgr.mpf.mp.br/solr/#/search/query?q=*:*&q.op=OR&indent=true&fq=search.resourceid:%2226c2fbee-a145-438d-9272-0d596bdb0b3b%22

[image: image.png]

Aroldo




Em ter., 19 de mar. de 2024 às 02:42, Alan Orth ***@***.***>
escreveu:

**Comment by aroldorique** [comment]
I am sending the image that were not attached correctly in my previous post.
![bug1](https://github.com/DSpace/DSpace/assets/31450466/0378be12-f477-4909-ba63-2d8462dc5525)

**Comment by tdonohue** [comment]
Related to #9660...and may be solved by any solution to that ticket.
